File: vol1/ch5/LeetCode202.py
class Solution:
    def isHappy(self, n: int) -> bool:
        hset=set()
        while n!=1 and n not in hset:
            hset.add(n)
            n=self.getnext(n)
        return n==1

    def getnext(self,m):
        ans=0
        while m>0:
            tmp=m%10
            ans+=tmp*tmp
            m=m//10
        return ans

File: vol1/ch5/LeetCode202.cpp
class Solution {
public:
    bool isHappy(int n) {
        unordered_set<int> hset;
        while (n!=1 && !hset.count(n)) {
            hset.insert(n);
            n=getnext(n);
        }
        return n==1;
    }
    int getnext(int m) {
        int ans=0;
        while(m>0) {
            int tmp=m%10;
            ans+=tmp*tmp;
            m=m/10;
        }
        return ans;
    }
};

